### PR TITLE
Set Temperature Slider Min/Max from device properties in Create Climate Setting Schedule form

### PR DESCRIPTION
**Pull request by Andrewjeska**
@xplato I changed the slider logic a bit to acommodate the seed data, I think it makes a bit more sense now, lemme know what you think. The previous logic gave the slider some weird behavior for `heat_cool` mode.

See weird behavior here: 

https://github.com/seamapi/react/assets/11382084/808300b1-6db8-4b4a-9fe3-f5f135d66102

### CODE REVIEW

File: .storybook/seed-fake.js [hunk 1]
```diff
@@ -416,18 +418,8 @@ export const seedFake = (db) => {
...
-        max_heating_set_point_fahrenheit: 79,
-        min_heating_set_point_fahrenheit: 45,
         device_name: 'debating book',
         ecobee_device_id: '14a7f34b-3e44-42b8-a314-ae953f81722a',
-        max_cooling_set_point_celsius: 33.333333333333336,
-        max_heating_set_point_celsius: 26.11111111111111,
-        min_cooling_set_point_celsius: 18.333333333333336,
-        min_heating_set_point_celsius: 7.222222222222222,
-        min_delta_heat_cool_set_points_celsius: 2.7777777777777777,
       },
```
Comment (Andrewjeska) [inline .storybook/seed-fake.js:423]: Updated seed data here

File: src/lib/temperature-bounds.ts [hunk 2]
```diff
@@ -23,34 +23,26 @@ export const getHeatBounds = ({
   mode,
   minHeat,
   maxHeat,
-  minCool,
```
Comment (razor-x) [inline src/lib/temperature-bounds.ts:26]: Why is `minCool` special?

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormClimateSetting.tsx [hunk 3]
```diff
@@ -29,6 +30,32 @@ export function ClimateSettingScheduleFormClimateSetting({
     device_id: deviceId,
   })
 
+  const properties = device?.properties as ThermostatDevice['properties']
```
Comment (razor-x) [inline src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormClimateSetting.tsx:33]:
```suggestion
if (!isThermostateDevice(device)) return null
const properties = device?.properties
```

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormClimateSetting.tsx [hunk 4]
```diff
@@ -29,6 +30,32 @@ export function ClimateSettingScheduleFormClimateSetting({
...
+
+  let setPointBounds = {}
+
+  if (properties.is_cooling_available) {
```
Comment (razor-x) [inline src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormClimateSetting.tsx:37]: Consider moving these to a function, `getSetPointBounds`

File: src/lib/temperature-bounds.ts (hunk 2 above)
Comment (Andrewjeska) [reply src/lib/temperature-bounds.ts:26]: Can you elaborate?

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormClimateSetting.tsx [hunk 5]
```diff
@@ -29,6 +30,34 @@ export function ClimateSettingScheduleFormClimateSetting({
     device_id: deviceId,
   })
 
+  if (device === undefined || device === null) return <></>
```
Comment (razor-x) [inline src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormClimateSetting.tsx:33]:
```suggestion
  if (device == null) return <></>
```

File: src/lib/temperature-bounds.ts (hunk 2 above)
Comment (Andrewjeska) [reply src/lib/temperature-bounds.ts:26]: Resolved over huddle, @xplato could use your eyes here for sure

File: src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormClimateSetting.tsx [hunk 6]
```diff
@@ -29,6 +30,34 @@ export function ClimateSettingScheduleFormClimateSetting({
...
 
+  if (device == null) return <></>
+  if (!isThermostatDevice(device)) return <></>
+  const properties = device.properties
+
+  let setPointBounds = {}
+
+  if (properties.is_cooling_available) {
+    setPointBounds = {
+      minCool: properties.min_cooling_set_point_fahrenheit,
```
Comment (xplato) [inline src/lib/ui/ClimateSettingForm/ClimateSettingScheduleFormClimateSetting.tsx:49]: Is this supposed to be `fahrenheit`?

**Review by xplato — APPROVED** [review]
Could be good to implement the `getSetPointBounds` function for better organization, but otherwise LGTM!

**Pull request merged**